### PR TITLE
Implement logger for multiprocessing.

### LINKED ISSUE #56: Log output adjustments
I would like to see more log output, in order to see that the script is actually doing.

Currently I only see:
``` 
Linkchecker instance started as expected.
Found site administrators email addresses for: m.leimgruber@4teamwork.ch
```

I would like to see something like:
```
${timestamp} - Start searching for broken internal links
${timestamp} - Found 203 broken internal links
${timestamp} - Start searching for external links
${timestamp} - Found 3012 external links
${timestamp} - Start checking external links
${timestamp} - ${error} received for ${link}
${timestamp} - ${error} received for ${link}
${timestamp} - ${error} received for ${link}
${timestamp} - ${error} received for ${link}
${timestamp} - Checked all external links
${timestamp} - Mail send to ${email}
```

### PR DESCRIPTION
**Pull request by busykoala**
Close #56  

This introduces a subclass of multiprocessing.Pool which has the ability to log from each process into a queue. Also it introduces a listener for the incoming logs.

TODO:
- [x] Log into std::out outside of multiprocessing
- [x] Check that logging while multiprocessing into std::out is not duplicate
- [x] Add/Change log message in multiprocessing worker

### CODE REVIEW

**Comment by busykoala** [comment]
@maethu, Could you please check out the multiprocessing logging? And is there a point in testing this setup?

File: ftw/linkchecker/linkchecker.py [hunk 1]
```diff
@@ -46,14 +50,19 @@ def get_uri_response(external_link_obj, timeout):
     return external_link_obj
 
 
-def limit_cpu():
-    os.nice(19)
-
-
 def work_through_urls(external_link_objs, timeout_config):
+    # get correct logger
+    logger = logging.getLogger('ftw.linkchecker')
```
Comment (maethu) [inline ftw/linkchecker/linkchecker.py]: Use consistent `LOGGER_NAME`

File: ftw/linkchecker/linkchecker.py [hunk 2]
```diff
@@ -46,14 +50,19 @@ def get_uri_response(external_link_obj, timeout):
...
+    logger_name = logger.name
```
Comment (maethu) [inline ftw/linkchecker/linkchecker.py]: Also `LOGGER_NAME`?

File: ftw/linkchecker/linkchecker.py [hunk 3]
```diff
@@ -46,14 +50,19 @@ def get_uri_response(external_link_obj, timeout):
...
+
+    # prepare worker function and pool
     part_get_uri_response = partial(get_uri_response, timeout=timeout_config)
-    pool = Pool(initializer=limit_cpu)
+    pool = PoolWithLogging(processes=cpu_count(), logger_name=logger_name)
```
Comment (maethu) [inline ftw/linkchecker/linkchecker.py]: `LOGGER_NAME`

File: ftw/linkchecker/linkchecker.py [hunk 4]
```diff
@@ -46,14 +50,19 @@ def get_uri_response(external_link_obj, timeout):
...
+
+    # run tasks using pool.map
```
Comment (maethu) [inline ftw/linkchecker/linkchecker.py]: You can remove that comment, doesn't help to understand the code better.

**Review by maethu — CHANGES_REQUESTED** [review]

File: ftw/linkchecker/linkchecker.py (hunk 1 above)
Comment (maethu) [reply ftw/linkchecker/linkchecker.py]: ?

File: ftw/linkchecker/linkchecker.py (hunk 2 above)
Comment (maethu) [reply ftw/linkchecker/linkchecker.py]: ?

File: ftw/linkchecker/linkchecker.py (hunk 3 above)
Comment (maethu) [reply ftw/linkchecker/linkchecker.py]: ?

File: ftw/linkchecker/linkchecker.py (hunk 4 above)
Comment (maethu) [reply ftw/linkchecker/linkchecker.py]: ?

**Review by maethu — CHANGES_REQUESTED** [review]

File: ftw/linkchecker/linkchecker.py (hunk 1 above)
Comment (busykoala) [reply ftw/linkchecker/linkchecker.py]: 👍

File: ftw/linkchecker/linkchecker.py (hunk 2 above)
Comment (busykoala) [reply ftw/linkchecker/linkchecker.py]: 👍

File: ftw/linkchecker/linkchecker.py (hunk 3 above)
Comment (busykoala) [reply ftw/linkchecker/linkchecker.py]: 👍

File: ftw/linkchecker/linkchecker.py (hunk 4 above)
Comment (busykoala) [reply ftw/linkchecker/linkchecker.py]: Yes thats absolutely true. It sometimes helps me create better visible sections so I forget that the comment itself does not add any value...

**Comment by busykoala** [comment]
@maethu, I now use the globally defined LOGGER_NAME everywhere. Also I got rid of the section where I get the logger to then get the logger name again.

**Review by maethu — APPROVED** [review]
yay 💛

**Pull request merged**